### PR TITLE
feat(cli): implement helm deployment list

### LINKED ISSUE #243: Client: List command
`helm list` List the deployments in the DM server

### PR DESCRIPTION
**Pull request by technosophos**
Closes #243

### CODE REVIEW

File: cmd/helm/helm.go [hunk 1]
```diff
@@ -30,6 +31,12 @@ func init() {
 	addCommands(cmds()...)
 }
 
+// Debug indicates whether the process is in debug mode.
+//
+// This is set at app start-up time, based on the presence of the --debug
+// flag.
+var Debug bool
```
Comment (adamreese) [inline cmd/helm/helm.go]: Why did you change this to a global?

File: cmd/helm/deployment.go [hunk 2]
```diff
@@ -17,18 +17,24 @@ limitations under the License.
...
 import (
+	"errors"
+	"regexp"
+
 	"github.com/codegangsta/cli"
+	"github.com/kubernetes/helm/pkg/format"
 )
 
 func init() {
+	//addCommands(listCmd())
```
Comment (adamreese) [inline cmd/helm/deployment.go]: Are you planning on adding a `helm list` "alias"

File: cmd/helm/deployment.go [hunk 3]
```diff
@@ -50,7 +56,46 @@ func deploymentCommands() cli.Command {
 				Name:      "list",
 				Usage:     "list all deployments, or filter by an optional pattern",
 				ArgsUsage: "PATTERN",
+				Action:    func(c *cli.Context) { run(c, list) },
 			},
 		},
 	}
 }
+
+func listCmd() cli.Command {
```
Comment (adamreese) [inline cmd/helm/deployment.go]: These functions names might need to be name spaced because of `repo list`, `cred list`, `chart list`, ...

File: cmd/helm/helm.go (hunk 1 above)
Comment (technosophos) [inline cmd/helm/helm.go]: I check it in `before()` now so that all of the commands can access it. Otherwise every command needs to parse the flag. Didn't need to make it public, though. I'll switch that to lowercase.

File: cmd/helm/deployment.go (hunk 2 above)
Comment (technosophos) [inline cmd/helm/deployment.go]: If people are okay with not having this alias to `helm list`, I want to remove this stuff. I think it's terribly confusing to have a `helm list` command, given that you can list several different things.
